docs/admin/troubleshooting_network.rst

```diff
@@ -2,3 +2,218 @@ Troubleshooting network issues
+
+   After a failed login, wait for a new two-factor code from your app before
+   trying again.
+
+You can reveal the passphrase by clicking the "eye" icon next to it in the login
+dialog (ensure you are in a fully private setting before doing so). Check for
+extra characters at the beginning at the end, or subtle differences like
+capitalization.
+
+If you use the two-factor app on your phone for other websites and services,
+make sure that you have selected the correct user account.
+
+If you have access to a Tails-based *Journalist Workstation*, verify whether you
+can access SecureDrop from Tails.
+
+If you are certain that your credentials are correct but you are unable to log
+in, proceed to the next step.
+
+Step 3: Verify that all required VMs are running
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+The following VMs must be running for all actions requiring network connectivity
+to work (e.g., logging in, checking for messages, downloading documents, replying
+to sources, starring sources, deleting sources):
+
+- ``sd-app``
+- ``sd-gpg``
+- ``sd-log``
+- ``sd-proxy``
+- ``sd-whonix``
+- ``sys-firewall``
+- ``sys-net``
+- ``sys-whonix`` (during updates)
+
+You can verify whether a VM is running or not by clicking the "Q" icon in the
+system tray (top right). Only VMs that are currently running will appear in the
+list:
+
+**[SCREENSHOT: Q widget with VM list]**
+
+If a required VM is not running, you can launch it from the Qube Manager. Open
+the Qube Manager by clicking **Open Qube Manager** in the menu above. A window
+like the following should appear:
+
+**[SCREENSHOT: Qube manager screenshot]**
+
+To start a VM, select it from the list, right-click it, and click **Start/Resume
+Qube**. Alternatively, you can click the "Play" button in the toolbar.
+
+In ordinary use, VMs required by SecureDrop should be started on boot or when
+they are needed. If you repeatedly experience problems with a necessary VM not
+running, or if an error message is displayed when attempting to start the VM,
+please contact us for assistance.
+
+If all required VMs are running, proceed to the next step.
+
+Step 4: Verify that required VMs have connectivity
+--------------------------------------------------
+In step 1, you have already verified that you can connect to the
+Internet using ``sys-net``. Now, test whether ``sys-firewall``, ``sd-whonix``
+and ``sd-proxy`` are working.
+
+First, open a terminal in ``sys-firewall`` and run the ``ping google.com`` command.
+You should see similar as in ``sys-net`` before.
+
+Now, open a terminal in ``sd-whonix`` and run the following command:
+
+``curl -s https://check.torproject.org/ | cat | grep -m 1 "Congratulations"``
+
+This command contacts a service intended for web browsers to verify whether your
+Tor connection is working.
+
+You should see the text "Congratulations. This browser is configured to use Tor."
+or a similar message on the terminal.
+
+If this command fails, proceed to the next step.
```

Comment:
> If this command fails

Several failure cases will be non-obvious, indicated by a _lack_ of a message, or a non-zero exit code. See:

```
user@host:~$ # Expecting a fully working connection
user@host:~$ curl -s https://check.torproject.org | grep -m 1 "Congratulations" 
      Congratulations. This browser is configured to use Tor.
user@host:~$ # After disconnecting network cable
user@host:~$ curl -s https://check.torproject.org | grep -m 1 "Congratulations" 
user@host:~$ echo $?
1
user@host:~$ sudo systemctl restart tor
user@host:~$ # After reconnecting network cable
user@host:~$ curl -s https://check.torproject.org | grep -m 1 "Congratulations" 
      Congratulations. This browser is configured to use Tor.
user@host:~$ # Stopping tor to simulate broken tor config
user@host:~$ sudo systemctl stop tor
user@host:~$ curl -s https://check.torproject.org | grep -m 1 "Congratulations" 
user@host:~$ echo $?
1
```

So consider stating explicitly:

> If this command does _not_ display a "Congratulations" message, proceed to the next step. 

or similar wording so that empty output doesn't appear to be a false negative.

Reply:
:+1: